File: db/src/main/java/org/delivery/db/userorder/UserOrderRepository.java
package org.delivery.db.userorder;

import java.util.List;
import java.util.Optional;
import org.delivery.db.userorder.enums.UserOrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;

/* 코틀린으로 변경
public interface UserOrderRepository extends JpaRepository<UserOrderEntity, Long> {

    // 특정 유저의 모든 주문내역 가져오기
    // select * from user_order where user_id = ? and status = ? order by id desc
    List<UserOrderEntity> findAllByUserIdAndStatusOrderByIdDesc(Long userId, UserOrderStatus status);

    // select * from user_order where user_id = ? and status = in(?, ?, ..) order by id desc
    List<UserOrderEntity> findAllByUserIdAndStatusInOrderByIdDesc(Long userId, List<UserOrderStatus> statusList);


    // 특정 주문 가져오기
    // select * from user_order where id = ? and status = ? and user_id = ?
    Optional<UserOrderEntity> findAllByIdAndStatusAndUserId(Long id, UserOrderStatus status, Long userId);

    // 특정 주문 가져오기 (상태를 제외한)
    // select * from user_order where id = ? and user_id = ?
    Optional<UserOrderEntity> findAllByIdAndUserId(Long id, Long userId);
}
*/

File: db/src/main/kotlin/org/delivery/db/userorder/UserOrderRepository.kt
package org.delivery.db.userorder

import org.delivery.db.userorder.enums.UserOrderStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserOrderRepository : JpaRepository<UserOrderEntity, Long> {
    // 특정 유저의 모든 주문내역 가져오기
    // select * from user_order where user_id = ? and status = ? order by id desc
    fun findAllByUserIdAndStatusOrderByIdDesc(userId: Long?, status: UserOrderStatus?): List<UserOrderEntity>

    // select * from user_order where user_id = ? and status = in(?, ?, ..) order by id desc
    fun findAllByUserIdAndStatusInOrderByIdDesc(
        userId: Long?,
        statusList: List<UserOrderStatus>?
    ): List<UserOrderEntity>

    // 특정 주문 가져오기
    // select * from user_order where id = ? and status = ? and user_id = ?
    fun findAllByIdAndStatusAndUserId(id: Long?, status: UserOrderStatus?, userId: Long?): UserOrderEntity?

    // 특정 주문 가져오기 (상태를 제외한)
    // select * from user_order where id = ? and user_id = ?
    fun findAllByIdAndUserId(id: Long?, userId: Long?): UserOrderEntity?
}
